pkgs/build-support/fetchurl/builder.sh

```diff
@@ -120,18 +125,17 @@ for url in $urls; do
             if test -n "${!varName}"; then mirrors="${!varName}"; fi
 
             for url3 in $mirrors; do
-                urls2="$urls2 $url3$fileName";
+                urls2+=("$url3$fileName")
             done
         fi
     fi
 done
-urls="$urls2"
```

Comment:
Could you keep the assignment, or perhaps `set -u` and delete the old array?
Having to use `url2` seems like a mistake waiting to happen.